R/utilities.r

```diff
@@ -299,6 +299,16 @@ find_args <- function(...) {
 # global data
 dummy_data <- function() data.frame(x = NA)
 
+with_seed <- function(seed, code) {
+  if (!is.null(seed)) {
+    old_seed <- get0(".Random.seed", globalenv())
```

Comment:
I think it is safe (I'd add `mode = 'integer'` to `get0()`, though), but it seems that you forgot to `set.seed(seed)` before `code` is evaluated?